### PR TITLE
Changed the elif notation to else if

### LINKED ISSUE #74: Rename elif to elsif
## Discussion

Currently, our "else if keyword" is `elif` which something I've derived from python, however

- Our language syntactically very much like ruby, and it would confuse the users (especially those who have a ruby background) and it's like learning a new language (since `elsif` is the ruby's way).
- On the other hand, if someone learned pocketlang, it would be easier for them to continue with ruby (so we'll keep it as closer to ruby as possible).
- We can also reuse any of ruby's syntax highlighter if we ever have to use one.
- Reading `elsif` sounds more like `else if` then `elif` does.

If you have any suggestions, regarding the renaming let me know.

## How to rename

- the name is defined at `src/pk_compiler.c` ([here is the line](https://github.com/ThakeeNathees/pocketlang/blob/c58159b63d5565a4788e300f5fb4e964b8d0c6d5/src/pk_compiler.c#L193))
- just renaming `"elif"` to `"elsif"` is sufficient but for consistency rename, it's token name from `TK_ELIF` to `TK_ELSIF` and find `elif` in the comments if any and replace them too.

### PR DESCRIPTION
**Pull request by ekinbarut**
Changed the `elif` notation to `else if` per requirements in #74 

*Edit Fix: #74*

### CODE REVIEW

File: src/pk_compiler.c [hunk 1]
```diff
@@ -129,7 +129,7 @@ typedef enum {
   TK_WHILE,      // while
   TK_FOR,        // for
   TK_IF,         // if
-  TK_ELIF,       // elif
+  TK_ELSEIF,     // else if
```
Comment (ThakeeNathees) [inline src/pk_compiler.c]: It should be `TK_ELSIF` and the keyword we're trying to rename to is `elsif`

File: src/pk_compiler.c [hunk 2]
```diff
@@ -190,7 +190,7 @@ static _Keyword _keywords[] = {
   { "while",    5, TK_WHILE    },
   { "for",      3, TK_FOR      },
   { "if",       2, TK_IF       },
-  { "elif",     4, TK_ELIF     },
+  { "else if",  4, TK_ELSEIF   },
```
Comment (ThakeeNathees) [inline src/pk_compiler.c]: it should be `"elsif"` and the 4 next to is the length of the keyword, which should be updated to 5 after the rename (and note that a keyword cannot contain a white space).

**Review by ThakeeNathees — CHANGES_REQUESTED** [review]
Thank you for taking your time and effort to make this PR, and welcome to pocketlang.

I've left a review on the pr for some minor changes it requirs.

File: src/pk_compiler.c (hunk 1 above)
Comment (ekinbarut) [reply src/pk_compiler.c]: I misunderstood the requirement, my bad. Changed the else if notation to elsif.

File: src/pk_compiler.c (hunk 2 above)
Comment (ekinbarut) [reply src/pk_compiler.c]: Changed the keyword count as well.

**Comment by ekinbarut** [comment]
Changed the keyword notation from else if to elsif, removed the whitespace, and updated the keyword count.

File: src/pk_compiler.c [hunk 3]
```diff
@@ -1078,7 +1078,7 @@ GrammarRule rules[] = {  // Prefix       Infix             Infix Precedence
   /* TK_WHILE      */   NO_RULE,
   /* TK_FOR        */   NO_RULE,
   /* TK_IF         */   NO_RULE,
-  /* TK_ELIF       */   NO_RULE,
+  /* TK_ELSIF     */   NO_RULE,
```
Comment (ThakeeNathees) [inline src/pk_compiler.c:1081]: Here the alignment is off by a white space, but that's okay, I'll merge it and fix it by myself. But it's in discussion at #74, I'd still like to hear from @xSavitar and @alexcpatel about the change and merge it by tomorrow.

Thanks!

Comment (ekinbarut) [reply src/pk_compiler.c:1081]: Thanks for the review. Good luck!

**Comment by xSavitar** [comment]
> Changed the `elif` notation to `else if` per requirements in #74

@ekinbarut, thanks for your change. It should be from `elif` to `elsif`. I mentioned a comment on the necessary task about the latter being very typo prone which is what I think your brain did here :]

Never the less, I think @ThakeeNathees, let's talk more about this a bit but I support the rename.

**Comment by ThakeeNathees** [comment]
@ekinbarut I'm really happy to merge this PR, but could you rename all the pocketlang unit tests where we're using the `elif` keyword. Because merging this currently will fail the build status.

Thanks in advance!

(here is the list of `elif` we've used, it would be helpful for you to rename. `fmt=path : line`)
````
.tests\examples\brainfuck.pk : 52
.tests\examples\brainfuck.pk : 57
.tests\examples\brainfuck.pk : 62
.tests\examples\brainfuck.pk : 67
.tests\examples\brainfuck.pk : 70
.tests\examples\brainfuck.pk : 76
.tests\examples\brainfuck.pk : 83
.tests\examples\brainfuck.pk : 90
.tests\examples\brainfuck.pk : 97
.tests\examples\fizzbuzz.pk : 5
.tests\examples\fizzbuzz.pk : 6
.tests\lang\controlflow.pk : 10
.tests\lang\controlflow.pk : 15
.tests\lang\controlflow.pk : 16
.tests\lang\controlflow.pk : 21
.tests\lang\controlflow.pk : 24
.tests\lang\controlflow.pk : 42
.tests\lang\controlflow.pk : 44
```

**Comment by ekinbarut** [comment]
Hello @ThakeeNathees, I renamed the keywords in the unit tests and saw that they were also used in prism and the 15-minute guide, so went ahead and changed those as well.

**Review by ThakeeNathees — APPROVED** [review]

**Comment by ThakeeNathees** [comment]
Thanks!

And we're glad to have you here, I'll open "good first issues" more If you're interested in this project and want to learn compiler principles together.

**Pull request merged**